### PR TITLE
enh(sec): get excluded users from password expiration policy

### PR DESCRIPTION
**Pull request by kduret**
## Description

get excluded users from password expiration policy

**Fixes** MON-12040

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 20.10.x
- [ ] 21.04.x
- [ ] 21.10.x
- [x] 22.04.x (master)

### CODE REVIEW

File: doc/API/centreon-api-v22.04.yaml [hunk 1]
```diff
@@ -3937,6 +3957,7 @@ components:
             - password_expiration
             - can_reuse_passwords
             - delay_before_new_password
+            - excluded_users
```
Comment (kduret) [inline doc/API/centreon-api-v22.04.yaml:3960]:
```suggestion
```

File: src/Core/Domain/Configuration/User/Model/User.php [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+        private bool $isAdmin,
+    ) {
+        Assertion::minLength($alias, self::MIN_ALIAS_LENGTH, 'User::alias');
+        Assertion::maxLength($alias, self::MAX_ALIAS_LENGTH, 'User::alias');
+
+        Assertion::minLength($name, self::MIN_ALIAS_LENGTH, 'User::name');
+        Assertion::maxLength($name, self::MAX_ALIAS_LENGTH, 'User::name');
+
+        Assertion::minLength($email, self::MIN_EMAIL_LENGTH, 'User::email');
+        Assertion::maxLength($email, self::MAX_EMAIL_LENGTH, 'User::email');
```
Comment (kduret) [inline src/Core/Domain/Configuration/User/Model/User.php:60]: assert email format

File: src/Core/Infrastructure/Security/ProviderConfiguration/Local/Repository/DbReadConfigurationRepository.php [hunk 3]
```diff
@@ -49,21 +49,62 @@ public function __construct(DatabaseConnection $db)
...
+            "SELECT c.`contact_id`, c.`contact_alias`, c.`contact_name`, c.`contact_email`, c.`contact_admin`
+            FROM `password_expiration_excluded_users` peeu
+            INNER JOIN `provider_configuration` pc ON pc.`id` = peeu.`provider_configuration_id`
+              AND pc.`name` = 'local'
+            INNER JOIN `contact` c ON c.`contact_id` = peeu.`user_id`
+              AND c.`contact_register` = 1"
+        );
+
+        $excludedUsers = [];
+        if ($statement !== false && $result = $statement->fetch(\PDO::FETCH_ASSOC)) {
```
Comment (kduret) [inline src/Core/Infrastructure/Security/ProviderConfiguration/Local/Repository/DbReadConfigurationRepository.php:103]:
```suggestion
        if ($statement !== false && $result = $statement->fetchAll(\PDO::FETCH_ASSOC)) {
```

File: src/Core/Infrastructure/Security/ProviderConfiguration/Local/Repository/DbReadConfigurationRepository.php [hunk 4]
```diff
@@ -49,21 +49,62 @@ public function __construct(DatabaseConnection $db)
...
+            $excludedUsers[] = $result;
+        }
```
Comment (kduret) [inline src/Core/Infrastructure/Security/ProviderConfiguration/Local/Repository/DbReadConfigurationRepository.php:105]:
```suggestion
        $excludedUsers = [];
        if ($statement !== false) {
            $excludedUsers = $statement->fetchAll(\PDO::FETCH_ASSOC);
        }
```

**Review by jeremyjaouen — APPROVED** [review]

**Review by adr-mo — APPROVED** [review]

**Pull request merged**